### PR TITLE
fix/intra run artifacts

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

## Commits

- fix(ci): pass build outputs between jobs as workflow artifacts, not cache
- fix: add additional sed commands to settings for improved documentation extraction
- fix: update SDK version drift message for clarity and accuracy in build workflow fix: correct output redirection in cache clearing workflow summary

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [ ] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: .github/workflows/_build.yaml [hunk 1]
```diff
@@ -94,30 +94,43 @@ jobs:
...
+                  echo "**Fix (CI):** trigger the [Refresh Lock Files]($REFRESH_WF_URL) workflow"
+                  echo "(scheduled every Monday 05:00 ET, or run manually via \`workflow_dispatch\`)."
+                  echo ""
+                  echo "**Fix (local):**"
+                  echo "\`\`\`"
+                  echo "sudo apt upgrade dotnet-sdk-10.0"
+                  echo "dotnet restore --force-evaluate"
+                  echo "\`\`\`"
+                  echo "Then re-push."
+              } >> "$GITHUB_STEP_SUMMARY"
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/workflows/_build.yaml:112]: commented a bit more

File: .github/workflows/_build.yaml [hunk 2]
```diff
@@ -94,30 +94,43 @@ jobs:
...
+      # this run. Artifacts -- not the cache -- are the intra-run handoff mechanism: the cache service is designed for
+      # cross-run reuse and gives no read-after-write guarantee, so a freshly saved entry may not be visible to a lookup
+      # seconds later (observed 2026-06-11: 4/4 deterministic restore misses ~25s after a verified save).
+      # The slug keeps artifact names unique across the build matrix legs (artifact names are immutable per run).
+      - name: Compute build artifact slug
+        id: slug
+        shell: bash
+        run: |
+          slug=$(echo '${{ inputs.build-project }}' | tr -c 'A-Za-z0-9.\n' '-')
+          echo "slug=${slug:-root}" >> "$GITHUB_OUTPUT"
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/workflows/_build.yaml:127]: @copilot, validate the changes please
